addons/mrp/models/stock_picking.py

```diff
@@ -50,6 +50,24 @@ class StockPickingType(models.Model):
         [('pdf', 'PDF'), ('zpl', 'ZPL')],
         "Generated Lot/SN Label to Print", default='pdf')
 
+    def _get_date_aggregated_records(
+        self, picking_types, model_name='stock.picking', required_state='assigned',
+        date_field_name='scheduled_date', data_series_name=_('Ready')
+    ):
+        mrp_picking_types = picking_types.filtered(lambda picking: picking.code == 'mrp_operation')
+        non_mrp_picking_types = (picking_types - mrp_picking_types)
+
+        records = super()._get_date_aggregated_records(
+            non_mrp_picking_types,
+            model_name, required_state, date_field_name, data_series_name
+        )
+        records.extend(super()._get_date_aggregated_records(
+            mrp_picking_types,
+            'mrp.production', 'confirmed', 'date_start', _('Confirmed')
```

Comment:
I would prefer to avoid to call `super()._get_date_aggregated_records` for empty recordset, because if you have both `repair` and `mrp` installed, you'll get something like that:

 - Call `mrp` `_get_date_aggregated_records`
   - Call `repair` `_get_date_aggregated_records` for `non_mrp_picking_types`
     - Call `stock` `_get_date_aggregated_records` for `non_repair_picking_types` (1) (returns Pickings data)
     - Call `stock` `_get_date_aggregated_records` for `repair_picking_types` (2) (returns Repair Orders data)
   - Call `repair` `_get_date_aggregated_records` for `mrp_picking_types`
     - Call `stock` `_get_date_aggregated_records` for `non_repair_picking_types` (3) (returns MO data)
     - Call `stock` `_get_date_aggregated_records` for `repair_picking_types` (4) (`picking_types` will always be empty)

And step (4) can be avoided

Reply:
That's a good idea :) But I can see that you've already changed my original code, is there anything left here to implement?